lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/util/AbstractCmdBuilder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runime.azbatch.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.reef.runtime.common.client.api.JobSubmissionEvent;
+import org.apache.reef.runtime.common.files.ClasspathProvider;
+import org.apache.reef.runtime.common.files.REEFFileNames;
+import org.apache.reef.runtime.common.launch.JavaLaunchCommandBuilder;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Abstract implementation of the OS command builder.
+ */
+public abstract class AbstractCmdBuilder implements CmdBuilder {
+
+  public static final String STD_OUT_FILE = "stdout.txt";
+  public static final String STD_ERR_FILE = "stderr.txt";
+
+  private final Class launcherClass;
+  private final List<String> commandListPrefix;
+  private final char classpathSeparatorChar;
+  private final String osCommandFormat;
+
+  protected final ClasspathProvider classpathProvider;
+  protected final REEFFileNames reefFileNames;
+
+  AbstractCmdBuilder(
```

Comment:
Can we expand Cmd to Command in this class name?

Reply:
Ugh! changed.